### PR TITLE
Update ready condition transition timestamp of ToolchainStatus only when status changed

### PR DESCRIPTION
**Pull request by alexeykazakov**

### CODE REVIEW

File: pkg/controller/toolchainstatus/toolchainstatus_controller_test.go [hunk 1]
```diff
@@ -74,6 +75,34 @@ func prepareReconcile(t *testing.T, requestName string, httpTestClient *fakeHTTP
 	return r, reconcile.Request{NamespacedName: test.NamespacedName(test.HostOperatorNs, requestName)}, fakeClient
 }
 
+func prepareReconcileWithStatusConditions(t *testing.T, requestName string, httpTestClient *fakeHTTPClient,
+	getMemberClustersFunc func(fakeClient client.Client) cluster.GetMemberClustersFunc, conditions []toolchainv1alpha1.Condition, initObjs ...runtime.Object) (*ReconcileToolchainStatus, reconcile.Request, *test.FakeClient) {
```
Comment (MatousJobanek) [inline pkg/controller/toolchainstatus/toolchainstatus_controller_test.go:79]: the `httpTestClient` as well as `getMemberClustersFunc` is not used in this function

File: pkg/controller/toolchainstatus/toolchainstatus_controller_test.go [hunk 2]
```diff
@@ -74,6 +75,34 @@ func prepareReconcile(t *testing.T, requestName string, httpTestClient *fakeHTTP
...
+	reconciler, req, fakeClient := prepareReconcile(t, requestName, newResponseGood(),
+		newGetMemberClustersFuncReady, initObjs...)
```
Comment (MatousJobanek) [inline pkg/controller/toolchainstatus/toolchainstatus_controller_test.go:81]: just a detail - can be on a single line: 
```suggestion
	reconciler, req, fakeClient := prepareReconcile(t, requestName, newResponseGood(), newGetMemberClustersFuncReady, initObjs...)
```

**Review by MatousJobanek — APPROVED** [review]
Nice :+1:

**Review by rajivnathan — APPROVED** [review]

**Comment by rajivnathan** [comment]
Thanks for fixing this 🙌

File: pkg/controller/toolchainstatus/toolchainstatus_controller.go [hunk 3]
```diff
@@ -408,21 +409,22 @@ func compareAndAssignMemberStatuses(reqLogger logr.Logger, toolchainStatus *tool
...
 	// any specific fields were updated. This way a problem with the controller can be indicated if the last updated timestamp was not updated.
-	conditionsWithTimestamps := []toolchainv1alpha1.Condition{}
-	for _, condition := range newConditions {
-		condition.LastTransitionTime = metav1.Now()
-		condition.LastUpdatedTime = &metav1.Time{Time: condition.LastTransitionTime.Time}
+	var conditionsWithTimestamps []toolchainv1alpha1.Condition
+	now := metav1.Now()
+	for _, newCondition := range newConditions {
+		condition := copyConditionWithLastTransitionTimeSet(status, newCondition)
+		condition.LastUpdatedTime = &now
```
[the remark this answers is not in the document]
Comment (alexeykazakov) [reply pkg/controller/toolchainstatus/toolchainstatus_controller.go:419]: Yeah.. we could do that. Will do it if the e2e tests fail! :)

[the remark this answers is not in the document]
Comment (alexeykazakov) [reply pkg/controller/toolchainstatus/toolchainstatus_controller.go]: Done in https://github.com/codeready-toolchain/host-operator/pull/332/commits/fefb77b76e9e977a40b243b9b150a7e241757b85

File: pkg/controller/toolchainstatus/toolchainstatus_controller_test.go (hunk 1 above)
Comment (alexeykazakov) [reply pkg/controller/toolchainstatus/toolchainstatus_controller_test.go]: Right. Cleaned it up in https://github.com/codeready-toolchain/host-operator/pull/332/commits/ebca00c3f55356cdfcda58e66fa42f542df28302

**Review by sbryzak — APPROVED** [review]

**Pull request merged**